\section{Methods}

\subsection{Study population}
57 patients with stable bronchiectasis were recruited as a part of this study by our collaborators in Milan, Italy. Recruitment of patients was performed cross-sectionally as a part of the Bronchiectasis Program of Fondazione IRCCS Ca' Granda Ospedale Maggiore Policlinico, Milan, Italy. Patients were enrolled during their clinical stability (at least one month apart from the last exacerbation and antibiotic course) and underwent clinical, radiological and microbiological evaluation. Patients were asked to provide a sputum and stool sample with a maximum gap of 12hrs. between the sputum and stool sample. DNA was extracted by our lab members from the sputum and stool samples as described previously \cite{Mac1800766}. The extracted DNA was subjected to targeted amplicon sequencing of the 16S rRNA and ITS2 regions of the genome to derive the Microbiome and Mycobiome, by mapping them to green genes and UNITE databases, respectively.

\subsection{Data-preprocessing}
Read counts of Microbiome and Mycobiome datasets from the sputum and stool samples of the 57 samples were converted into relative abundances. Only microbes present $\geq$1\% in atleast 5 patients were considered for further analysis. 

\subsection{Co-occurrence analysis}


\subsection{}

We report the first integration of lung-gut microbiome (bacteria and fungi) in bronchiectasis (n=57) perfomed using MOFA2 and a weighted similarity network fusion approach following spectral clustering. ALDEX2 was implemented to identify discriminant taxa. Co-occurence analysis was employed using GBLM to dervie microbial association networks.